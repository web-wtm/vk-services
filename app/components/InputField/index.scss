.field-group {
    vertical-align: middle;
    display: inline-block;

    label {
        font-size: 28px;
        padding-right: 10px;
    }
    
    input {
        font-size: 18px;
        padding: 10px;
        font-family: 'Orbitron', sans-serif;
        line-height: 1;
        &:focus {
            outline: none;
        }
    }
}