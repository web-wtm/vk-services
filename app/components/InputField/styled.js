import styled from 'styled-components'

export const FieldGroup = styled.div`
    vertical-align: middle;
    display: inline-block;
`

export const Label = styled.label`
    font-size: 28px;
    padding-right: 10px;
`

export const Input = styled.input`
    font-size: 18px;
    padding: 10px;
    font-family: 'Orbitron', sans-serif;
    line-height: 1;
    &:focus {
        outline: none;
    }
`