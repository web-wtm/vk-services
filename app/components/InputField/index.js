import React from 'react'
import { Input } from './styled'

const InputField =({fieldName, placeHolder, value, type, label, onChange}) => {
    return (
        <Input 
            name={fieldName}
            value={value}
            placeholder={placeHolder}
            type={type}
            onChange={onChange}
        />
    )
}

InputField.defaultProps = {
    type: 'text',
    placeHolder: ''
}

export default InputField;