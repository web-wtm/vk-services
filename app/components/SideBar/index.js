import styled from 'styled-components'

const SideBar = styled.div`
    background-color: #222;
    font-family: 'Podkova',serif;
    position: fixed;
    top: 90px;
    left: 0;
    z-index: 1;
    height: 100%;
    width: 270px;

    @media (max-width: 600px) {
        width: 160px;
    }
`

export default SideBar