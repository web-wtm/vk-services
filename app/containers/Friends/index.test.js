import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Friends } from './index'
import SideBar from '../../components/SideBar'
import InputField from '../../components/InputField'

describe('Friends component', () => {
    const store = {
        friends: {
            mutualFriendsId: []
        }
    };

    const component = shallow(<Friends state={store} getUserId={() => {}} />);

    it('rendered Friends component', () => {
        const component = renderer.create(<Friends state={store} getUserId={() => {}} />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('rendered inputs', () => {
        const redneredSideBar = component.find(SideBar);
        expect(redneredSideBar.find(InputField)).toHaveLength(3)
    });
});