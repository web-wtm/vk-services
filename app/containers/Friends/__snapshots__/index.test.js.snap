// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friends component rendered Friends component 1`] = `
Array [
  <div
    onClick={[Function]}
    style={
      Object {
        "bottom": 50,
        "cursor": "pointer",
        "opacity": 0,
        "position": "fixed",
        "right": 30,
        "transitionDelay": "0s",
        "transitionDuration": "0.2s",
        "transitionProperty": "opacity, visibility",
        "transitionTimingFunction": "linear",
        "visibility": "hidden",
        "zIndex": 5,
      }
    }
  >
    <span
      className="sc-bdVaJa iyrqmb"
    >
      up
    </span>
  </div>,
  <div
    className="sc-bxivhb ewfoPr"
  >
    Enter users id to know their mutual friends (you can find out user id below)
  </div>,
  <div
    className="sc-bZQynM dcysoQ"
  >
    <input
      className="sc-EHOje dzcabn"
      name="source_uid"
      onChange={[Function]}
      placeholder="source user id"
      type="text"
    />
    <input
      className="sc-EHOje dzcabn"
      name="target_uid"
      onChange={[Function]}
      placeholder="target user id"
      type="text"
    />
    <button
      className="sc-ifAKCX bszNgu"
      onClick={[Function]}
    >
      check
    </button>
    <input
      className="sc-EHOje dzcabn"
      name="user_ids"
      onChange={[Function]}
      placeholder="short name"
      type="text"
    />
    <button
      className="sc-ifAKCX bszNgu"
      onClick={[Function]}
    >
      check id
    </button>
  </div>,
]
`;
